test/server/config/stats/config_test.cc

```diff
@@ -0,0 +1,89 @@
+#include <string>
+
+#include "envoy/registry/registry.h"
+
+#include "common/protobuf/utility.h"
+#include "common/stats/statsd.h"
+
+#include "server/config/stats/statsd.h"
+
+#include "test/mocks/server/mocks.h"
+
+#include "api/bootstrap.pb.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+using testing::NiceMock;
+using testing::Return;
+using testing::ReturnRef;
+using testing::_;
+
+namespace Server {
+namespace Configuration {
+
+TEST(StatsConfigTest, ValidTcpStatsd) {
+  const std::string name = "envoy.statsd";
+  Protobuf::Struct config;
+  ProtobufWkt::Map<ProtobufTypes::String, ProtobufWkt::Value>& field_map = *config.mutable_fields();
```

Comment:
Prefer using `auto` here and elsewhere for proto handlers, the extra noise of the complex types doesn't help readability.